### ISSUE TITLE
Cannot restore the cluster from S3 backups 

### ISSUE THREAD

**Issue by Radeeka** [CrunchyData/postgres-operator#3457]
We have a Postgres cluster running on Kubernetes, setup using PGO and S3 is used as the backup option. The cluster and backup operation are working perfectly fine. 

Cluster name: database-1
Namespace: db-new

And now I want to clone this setup in to a new cluster and I'm hoping to use the S3 as the backup storage in the new cluster as well. (using the same bucket).

New Cluster name: database-2
Namespace: db-new-2

I've gone through many issues in this repo and finally came up with the following cluster creation command.

pgo create cluster database-2 -n db-new-2 \
--replica-count=1 \
--pgbackrest-s3-bucket <bucket name> \
--pgbackrest-s3-endpoint s3.<region>.amazonaws.com \
--pgbackrest-s3-key  <key> \
--pgbackrest-s3-key-secret <secret> \
--pgbackrest-s3-region <region> \
--pgbackrest-storage-type s3 \
--pgbackrest-s3-ca-secret <kubernetes secret> \
--restore-from database-1 \
--restore-from-namespace db-new \
--restore-opts="--repo-type=s3" \
--pgbouncer 

But I'm getting the following error once the bootstrap pod starts. 

___________________________________________________________________________________________________________________________________________

2022-11-15 09:51:39,091 INFO: Lock owner: None; I am database-2-bootstrap-sgmm9
2022-11-15 09:51:39,092 INFO: not healthy enough for leader race
2022-11-15 09:51:39,092 INFO: bootstrap in progress
WARN: repo1: [HostConnectError] unable to load info file '/backrestrepo/database-1-backrest-shared-repo/backup/db/backup.info' or '/backrestrepo/database-1-backrest-shared-repo/backup/db/backup.info.copy':
      HostConnectError: raised from remote-0 protocol on 'database-1-backrest-shared-repo.db-new.svc.cluster.local.': timeout connecting to '[bucket-name].s3.[region].amazonaws.com:443'
      HostConnectError: raised from remote-0 protocol on 'database-1-backrest-shared-repo.db-new.svc.cluster.local.': timeout connecting to '[bucket-name].s3.[region].amazonaws.com:443'
      HINT: backup.info cannot be opened and is required to perform a backup.
      HINT: has a stanza-create been performed?
ERROR: [075]: no backup set found to restore
Tue Nov 15 09:51:44 UTC 2022 ERROR: pgBackRest primary Creation: pgBackRest restore failed when creating primary
2022-11-15 09:51:44,507 INFO: removing initialize key after failed attempt to bootstrap the cluster
2022-11-15 09:51:44,727 INFO: renaming data directory to /pgdata/database-2_2022-11-15-09-51-44
Traceback (most recent call last):
  File "/usr/local/bin/patroni", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 170, in main
    return patroni_main()
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 138, in patroni_main
    abstract_main(Patroni, schema)
  File "/usr/local/lib/python3.6/site-packages/patroni/daemon.py", line 100, in abstract_main
    controller.run()
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 108, in run
    super(Patroni, self).run()
  File "/usr/local/lib/python3.6/site-packages/patroni/daemon.py", line 59, in run
    self._run_cycle()
  File "/usr/local/lib/python3.6/site-packages/patroni/__init__.py", line 111, in _run_cycle
    logger.info(self.ha.run_cycle())
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1457, in run_cycle
    info = self._run_cycle()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1351, in _run_cycle
    return self.post_bootstrap()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1247, in post_bootstrap
    self.cancel_initialization()
  File "/usr/local/lib/python3.6/site-packages/patroni/ha.py", line 1240, in cancel_initialization
    raise PatroniFatalException('Failed to bootstrap cluster')
patroni.exceptions.PatroniFatalException: 'Failed to bootstrap cluster'
___________________________________________________________________________________________________________________________________________

It says that the /backrestrepo/database-1-backrest-shared-repo/backup/db/backup.info file is not available. But its available in the very location in the S3 bucket. 
![image](https://user-images.githubusercontent.com/21239899/201898131-bc7bd303-fd7b-4566-8c1d-afeb18442ef5.png)

Backups have also been performed without any issue.
![image](https://user-images.githubusercontent.com/21239899/201898805-02e178f7-acfa-4029-bfeb-49eb1b382f47.png)

pgo client version 4.7.0
pgo-apiserver version 4.7.0

Any help in getting through this is highly appreciated. Thanks in advance!

**Comment by dsessler7** [comment]
@Radeeka, are you still experiencing this issue? The errors imply that the file is "not available" because the connection is timing out when trying to connect to the host, so it seems like a network issue. Are you certain you have the URI, endpoint, region, etc specified correctly? Can you try this with the latest v4 version, or better yet, v5?

**Comment by Radeeka** [comment]
> @Radeeka, are you still experiencing this issue? The errors imply that the file is "not available" because the connection is timing out when trying to connect to the host, so it seems like a network issue. Are you certain you have the URI, endpoint, region, etc specified correctly? Can you try this with the latest v4 version, or better yet, v5?

**Comment by Radeeka** [comment]
@dsessler7 Never found a fix for this particular approach and I'm pretty sure the parameters were accurately specified. Also we had to stick with the same PGO version due to some other dependencies. However I ended up implementing this with a different approach. 
If anyone else having the same issue, this is how I managed. 
Created a new cluster and replaced the backrestrepo directory contents with the S3 backup of the initial cluster. Then created a new cluster using above mentioned cluster and it worked for me. 
Note: 've tried this with PGO 4.7 and it worked. Also make sure you change the directory ownership recursively when you copy the archive and backup directories to the new repo directory to 2000 (or the UID of the user which creates the backrestrepo directory inside the pgbackrest pod) or else the pgbackrest process won't be able to read the contents inside during the cloning stage.
